txp/Pos/Binary/Txp/Core.hs

```diff
@@ -19,10 +19,20 @@ import qualified Pos.Txp.Core.Types as T
 ----------------------------------------------------------------------------
 
 instance Bi T.TxIn where
-  encode txIn = encodeListLen 2 <> encode (T.txInHash txIn) <> encode (T.txInIndex txIn)
-  decode = do
-    enforceSize "TxIn" 2
-    T.TxIn <$> decode <*> decode
+    encode T.TxInUtxo{..} =
+        encodeListLen 2 <>
+        encode (0 :: Word8) <>
+        encode (serialize' (txInHash, txInIndex))
```

Comment:
Is using `serialize'` inside `encode` legal at all? Why not `encodeListLen 3 <> ...`  instead?